% !Rnw root = sturgeon-fluidigm-main.Rnw

\section*{Introduction}

This is super free-form, stream of consciousnessy at the moment, as I get some ideas
down onto the page.

The study of highly migratory species in environments that are difficult to monitor present
a number of challenges.   Often, the migrations themselves cannot be observed.
Genetic data has been successful in inferring the ecology of a number of species
that are otherwise difficult to study (CITE).  It is now fairly routine to do this
in most species, especially diploid ones.  Genetic methodologies are well-established for microsats, SNPs, and to an increasing extent, lately, next generation sequencing data.
Concurrently, the last two decades have seen an explosion in the availability statistical
methods for analyzing genetic data.  

The vast majority of methods have been developed for diploid organisms.  When these methods
can be extended to polyploids their extension
typically requires that the number of copies of individual alleles can be resolved in a
polyploid genotype, or that the exact inheritance patterns are known, which is not always
the case.  The challenges of polyploids have been
well-appreciated in botanical fields for some time (CITE Fisher's polyploid inheritance
paper), and these difficulties have been gaining increasing attention in the field of
molecular ecology (CITE that recent review).  

Significant advances have been made in the development of models for polyploids and
particularly for tetraploids, for example, fitTetra, and the work from Brazil on
sugar cane.  However, sometimes that quality of scoring is not high enough to 
resolve all those allele copies. Furthermore, the assumptions underylying fitTetra
are probably not appropriate when samples are mixture of diverged tetraploid populations.
(ERIC VERIFY THIS)
Clearly, it would be best to be able to resolve
and count all the alleles, but in cases where that is not possible for a majority of the
loci that have been discovered, alternative modes of analysis are sought.  One such
alternative analysis method is to treat alleles in polyploids as dominant markers.  This
approach has been used recently in population assignment (Israel et al.) and CITE, and CITE,
and more recently Wang and Scribner have shown the utility of that approach for doing
relationship inference.  

Modeling everything as a dominant marker (presence/absence of the allele) is particularly
well-suited to microsatellite loci where length polymorphisms are easy to interpret.
However, using xxxxxx-type SNP assays (like fluidigm assays)  the individual alleles
may not be so easily resolved, because the scoring relies on individual genotypes clustering
in two space, and it might not always be easy to resolve the 5 expected genotype clusters.
For such data, trying to resolve the data into individual, expected genotype categories
might be unworkable for many loci, especially if typing members of fairly well-diverged
lineages.  Depending on the goal of genetic analysis, however, it may still not be
necessary to require thinking about the data only as  individual alleles giving rise to
a full complement of expected genotypes.



\section*{Methods}



\subsection*{Genetic Samples}

We received samples of two general categories.  Samples collected in freshwater rivers known to have 
well-established spawning populations of green sturgeon---Sacramento River and Klamath Rivers in this study---are
``reference'' samples. Some of these samples were taken from eggs collected on egg mats or 
from juvenile fish, either of which must have been spawned in the river from which they were sampled. 
Still over reference samples were 
taken from adult fish that may be non-spawning visitors that are actually native to a river different 
from the one they were sampled in.  We refer to these groups as ``egg/juvenile
reference'' and ``adult reference'' 
samples, respectively.  Samples taken from the ocean or from rivers without an historically abundant 
spawning population are ``non-reference'' samples.  They are comprised of green sturgeon caught as 
bycatch, and five adult fish sampled from the Eel River, CA (Table~\ref{tab:samps}).
\begin{table*}
\caption{Summary of genetic samples used in this study.}
\begin{center}
\begin{tabular}{llllr}
\hline \hline \\ 
\input{../outputs/sample_summary.tex}
\hline \hline \\ 
\end{tabular}
\end{center}
\label{tab:samps}
\end{table*}
Tissue samples were predominantly fin clips dried or stored in ethanol, apart from the eggs which were whole developing eggs, stored in ethanol. 



\subsection*{SNP Discovery and Assay Development}

In order to discover SNPs to be turned into Fluidigm SNP-type assays,  we first sequenced eight fish
using a double digest restriction-site associated DNA protocol
(ddRAD; \citealt{Petersonetal2012})  on a MiSeq sequencer (Illumina Inc., San Diego, CA).
This protocol uses restriction enzyme digestion and a size selection step to reduce the fraction of
genome that is sequenced, so that the different individuals can be sequenced to a high depth at a common set of genomic locations.  Four of the samples were taken from screw traps in the Klamath River (Northern DPS) and four originated from the Upper Sacramento River. 21.24 million sequence reads of
XXXX base pairs generated by the sequencing run were assembled into 84,747 putatively unique genetic loci (regions of the genome) using Stacks software v 1.12 \citep{catchen2011stacks}. We queried the resulting database for loci containing single nucleotide polymorphisms (SNPs) that appeared diagnostic for each DPS (WHAT DOES THIS MEAN?), identifying 1118 such SNPs. Because the green sturgeon has a tetraploid genome, it is difficult to verify diagnostic capability of a given SNP from a sample of 8 individuals. Therefore, after filtering this list of potential SNPs for quality, we submitted the best 96 of them to Fluidigm Inc. (South San Francisco, CA) for SNPtype assay design.

The 96 SNPtype assays were loaded together with 94 green sturgeon samples and two non-template controls onto four 96.96 Fluidigm dynamic arrays for genotyping. Fluidigm nanofluidic technology allows simultaneous genotyping of all 9,216 sample-by-locus combinations.  Resultant raw fluorescence data from samples of known origin were submitted to FitTetra package for R \citep{voorrips2011genotype}, which fits mixture models to call tetraploid loci, allowing us to call genotypes in 49 of the 96 loci. Of these 49 loci, 29 had genetic differentiation greater than 1\% between known samples from the Northern and Southern DPS as measured by the GST statistic in Genodive v2.0b25 \citep{meirmans2004genotype}. We then manually curated the genotype data, selecting 12 loci for use in Genetic Stock Identification based on their relative ease of scoring. Genotypes for these 12 loci from 21 Northern DPS and 39 Southern DPS individuals comprise the genetic baseline of known samples that we used in subsequent GSI analyses of bycatch. Raw fluorescence data from 152 bycatch samples were then scored using the mixture discriminant analysis function in the mda package for R \citep{hastie1996discriminant}, followed by manual curation. Genotype data from each of two sets of bycatch samples were submitted as separate fisheries mixtures to gsisim \citep{Andersonetal2008} for population assignment against the genetic baseline.

\section*{Results}

\section*{Discussion}

\section*{Acknowledgments}
We are grateful to \ldots