\documentclass[twoside,10pt,twocolumn]{article}

% this stuff just has to be in here so it doesn't choke 
% on the stuff that I put in for typesetting the submission version.
\newcommand{\AuthorAddresses}{}
\newcommand{\KeyWords}{}
\newcommand{\CorrespondingAuthor}{}
\newcommand{\RunningTitle}{}


\input{author-title-etc}  % Put all the author title stuff in there

% ------
% Fonts and typesetting settings
\usepackage[sc]{mathpazo}
\usepackage[T1]{fontenc}
\linespread{1.05} % Palatino needs more space between lines

\usepackage{microtype}
% ------
% Page layout
\usepackage[labelfont=bf,labelformat=simple,labelsep=quad]{caption}

\usepackage{graphicx}
\usepackage{amssymb}
\usepackage{epstopdf}
\usepackage{amsfonts}
\usepackage{natbib}
\usepackage{subfigure}
\usepackage{xspace}
\usepackage{mathrsfs}
\usepackage{fancyhdr}
\usepackage{cuted}
\usepackage{flushend}
\usepackage{color}



%% some handy things for making bold math
\def\bm#1{\mathpalette\bmstyle{#1}}
\def\bmstyle#1#2{\mbox{\boldmath$#1#2$}}
\newcommand{\thh}{^\mathrm{th}}


%% Some pretty etc.'s, etc...
\newcommand{\cf}{{\em cf.}\xspace }
\newcommand{\eg}{{\em e.g.},\xspace }
\newcommand{\ie}{{\em i.e.},\xspace }
\newcommand{\etal}{{\em et al.}\ }
\newcommand{\etc}{{\em etc.}\@\xspace}

\newcommand{\snptype}{SNP~Type\textsuperscript{{\tiny TM}}}
\newcommand{\bp}{\bm{p}}


%% the page dimensions
\textwidth = 6.5 in
\textheight = 9 in
\oddsidemargin = -0.01 in
\evensidemargin = -0.01 in
\topmargin = -0.7 in
\headheight = 0.25 in
\headsep = 0.25 in
\parskip = 0.0in
\parindent = 0.25in

\setlength{\columnsep}{.4in}


%% change section heading styles
\makeatletter
\renewcommand\section{\@startsection {section}{1}{\z@}%
                                   {-3.5ex \@plus -1ex \@minus -.2ex}%
                                   {1.5ex \@plus 0ex}%
                                   {\normalfont\normalsize\bfseries}}
\renewcommand\subsection{\@startsection{subsection}{2}{\z@}%
                                     {-3.25ex\@plus -1ex \@minus -.2ex}%
                                     {1.5ex \@plus .2ex}%
                                     {\normalfont\normalsize\itshape}}
\makeatother

%% modify the abstract environment
\renewenvironment{abstract}
{\begin{quote}{\bf Abstract}\end{quote}\begin{quote}\bfseries\small}
{\end{quote}}


\fancypagestyle{firststyle}
{
   \fancyhf{}
   \chead[]{\vspace*{.1in}\includegraphics[width=\textwidth]{images/banner.pdf}}
   \lfoot[]{\footnotesize \myCopyright}
}

%% here is what I hope will be fancyplain
\fancyhead{} % clear all header fields
\fancyhead[LE]{{\bf \thepage}~~{\sl \myRunningTitle}}
\fancyfoot[RE,LO]{{\footnotesize \myCopyright}}
\renewcommand{\headrulewidth}{0pt}
\fancyhead[RO]{{\sl \myRunningAuthor}~~{\bf \thepage}}
\cfoot[]{}


%%%%%%%% KNITR BLOCK
% Here is a knitr chunk to set global values
<<setup, include=FALSE, cache=FALSE>>=
docDir <- getwd()

# set global chunk options
knitr::opts_chunk$set(include = FALSE, cache = FALSE)
knitr::opts_knit$set(root.dir = file.path(docDir, ".."))
#knitr::opts_knit$set(self.contained = FALSE)
knitr::opts_knit$set(concordance = TRUE)
@



% a knitr chunk to evaluate the code needed to produce variables and figures we need
% in this section
<<set-flags-and run R-main scripts, include=FALSE, cache = FALSE>>=
ReDoStructureRuns = FALSE

# these scripts write output to the outputs directory
# and can be commented out for faster compiling of the paper
source("R-main/00-meta-data-summaries.R")
#source("R-main/01-develop-chip-scoring.R")
#source("R-main/01.5-score-plate-five.R")
source("R-main/03-prep-for-structure.R")
#source("R-main/05-indiv-id-sims.R")
@



\begin{document}


\pagestyle{fancyplain}
\thispagestyle{firststyle}



   \begin{strip}
   \mbox{}\\
        {\LARGE\bf \myTitle \par}
    \mbox{}\\
    \uppercase{\myAuthors}\\ 
       \mbox{}\\
    {\em \myAffiliations}\\
    \mbox{}\\
    {\small \myEmailFootnote Correspondence: \myEmailAddress}
    
%   \begin{abstract}
%   	\input{abstract.Rnw}     %% Put abstract text in here.  If we want to include R code in it we will have to
                             %% use Sexpr{knit_child("doc.Rnw")} instead of \input{abstract.Rnw}.  But using \input gives us good
                             %% syncing
%    \end{abstract}
   \end{strip}

{\small
% here, input the main body text part.
%<<main-body-insert, child="main-body-text.Rnw">>=
%@
%% I grew weary of not having PDF sync, so I just inserted the whole file in here.  That
%% Makes it harder to do the one column version, but I can probably figure out a workaround
%% By just processing the file with awk and putting this into it.
%% It is a shame that Yihui doesn't want to support concordance with child documents.
%%%%%%%%%%%%%%%% START MAIN BODY

\section*{Introduction}

The Southwest Fisheries Science Center's (SWFSC) Fisheries Ecology Division (FED) in Santa Cruz, CA
received tissue samples from adult green sturgeon caught as bycatch in West Coast fisheries
from 2007 to 2013.  The tissues were provided by the Southwest Region with the intent
that the SWFSC develop a cost-effective, single nucleotide polymorphism (SNP)
genotyping method to identify the Distinct Population Segment that each
sample belonged to, and, if possible, to identify DNA samples 
from the same individual, thus potentially allowing the detection of fish that had been captured
(and sampled) more than once as bycatch.

The SWFSC also obtained tissue samples from juvenile sturgeon and sturgeon eggs native to the
Klamath and Sacramento Rivers---spawning grounds for Northern and Southern DPS fish, respectively.
These were used to identify genetic differences between the two DPSs in a genome scan, to develop
SNP assays, and for use in a genetic baseline to assigning fish to DPS.

As sturgeon are a tetraploid species, they are not amenable to SNP genotyping using
all of the technologies developed for genotyping diploid species.  Our work required
developing genotyping system and workflow to allow scoring of tetraploid species using
the Fluidigm \snptype{} assay system.  In our system, we do not resolve the individual
alleles and their dosages carried by each tetraploid individual, but, rather, score individuals
into ``genotype categories.''  The inheritance properties of such ``genotype categories'' are
not known, but with proper tailoring of the statistical models involved they are still
suitable for assessment of population structure, assigment of 
individuals to DPS, and identification of samples from the same indvidual.

We describe our methods and results below.  The results make it clear that there is a substantial
amount of genetic differnentiation between the two DPSs, and that the panel of SNPs and
the statistical procedures we have developed
provide a robust, cost-effective, and extremely accurate means of identifying the DPS of 
individual green sturgeon and for identifying green sturgeon that have been encountered
and sampled multiple times.






%%%%% HERE ARE THOUGHTS FOR AN INTRODUCTION FOR A JOURNAL PAPER  %%%%%%%%%%%
% This is super free-form, stream of consciousnessy at the moment, as I get some ideas
% down onto the page.
% 
% The study of highly migratory species in environments that are difficult to monitor present
% a number of challenges.   Often, the migrations themselves cannot be observed.
% Genetic data has been successful in inferring the ecology of a number of species
% that are otherwise difficult to study (CITE).  It is now fairly routine to do this
% in most species, especially diploid ones.  Genetic methodologies are well-established for microsats, SNPs, and to an increasing extent, lately, next generation sequencing data.
% Concurrently, the last two decades have seen an explosion in the availability statistical
% methods for analyzing genetic data.  
% 
% The vast majority of methods have been developed for diploid organisms.  When these methods
% can be extended to polyploids their extension
% typically requires that the number of copies of individual alleles can be resolved in a
% polyploid genotype, or that the exact inheritance patterns are known, which is not always
% the case.  The challenges of polyploids have been
% well-appreciated in botanical fields for some time (CITE Fisher's polyploid inheritance
% paper), and these difficulties have been gaining increasing attention in the field of
% molecular ecology (CITE that recent review).  
% 
% Significant advances have been made in the development of models for polyploids and
% particularly for tetraploids, for example, fitTetra, and the work from Brazil on
% sugar cane.  However, sometimes that quality of scoring is not high enough to 
% resolve all those allele copies. Furthermore, the assumptions underylying fitTetra
% are probably not appropriate when samples are mixture of diverged tetraploid populations.
% (ERIC VERIFY THIS)
% Clearly, it would be best to be able to resolve
% and count all the alleles, but in cases where that is not possible for a majority of the
% loci that have been discovered, alternative modes of analysis are sought.  One such
% alternative analysis method is to treat alleles in polyploids as dominant markers.  This
% approach has been used recently in population assignment (Israel et al.) and CITE, and CITE,
% and more recently Wang and Scribner have shown the utility of that approach for doing
% relationship inference.  
% 
% Modeling everything as a dominant marker (presence/absence of the allele) is particularly
% well-suited to microsatellite loci where length polymorphisms are easy to interpret.
% However, using xxxxxx-type SNP assays (like fluidigm assays)  the individual alleles
% may not be so easily resolved, because the scoring relies on individual genotypes clustering
% in two space, and it might not always be easy to resolve the 5 expected genotype clusters.
% For such data, trying to resolve the data into individual, expected genotype categories
% might be unworkable for many loci, especially if typing members of fairly well-diverged
% lineages.  Depending on the goal of genetic analysis, however, it may still not be
% necessary to require thinking about the data only as  individual alleles giving rise to
% a full complement of expected genotypes.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\section*{Methods}



\subsection*{Genetic Samples}

We received samples of two general categories.  Samples collected in freshwater rivers known to have 
well-established spawning populations of green sturgeon---Sacramento River and Klamath River, in this study---are
``reference'' samples. Some of these samples were taken from eggs collected on egg mats or 
from juvenile fish, either of which must have been spawned in the river from which they were sampled. 
We refer to these as ``egg/juvenile reference'' samples.  By contrast, ``non-juvenile reference'' samples are 
taken from adult fish that may be non-spawning visitors, native to a river different 
from the one they were sampled in. 
Samples taken from the ocean or from rivers without an historically abundant 
spawning population are ``non-reference'' samples.  They are comprised of green sturgeon caught as 
bycatch in ocean fisheries, and five adult fish sampled from the Eel River, CA.   See Table~\ref{tab:samps} for a summary
of sample sizes.
\begin{table*}
\caption{Summary of genetic samples used in this study.}
\begin{center}
\begin{tabular}{llllr}
\hline \hline \\ 
\input{../outputs/sample_summary.tex}
\hline \hline \\ 
\end{tabular}
\end{center}
\label{tab:samps}
\end{table*}
Tissue samples were predominantly fin clips, dried or stored in ethanol. Samples of eggs were whole, developing eggs, stored in ethanol, and the Klamath River reference samples, were whole, small juveniles stored in ethanol. 



\subsection*{SNP Discovery and Assay Development}

To discover SNPs to be made into Fluidigm \snptype{} assays,  we first sequenced eight fish
using a double-digest restriction-site associated DNA protocol
(ddRAD; \citealt{Petersonetal2012})  on a MiSeq sequencer (Illumina Inc., San Diego, CA).
This protocol uses restriction enzyme digestion and DNA fragment size selection
to reduce the fraction of
genome that is sequenced, ensuring that a small portion of the genome can be sequenced to
a sufficiently high read depth in a number of different individuals.  Four of the samples were
juveniles from the Klamath River (Northern DPS) and four were juveniles from the
Upper Sacramento River. 21.24 million sequence reads, each of
300 base pairs, generated by the sequencing run were assembled into 84,747 putatively unique genetic 
loci (regions of the genome) and, within those regions, we identified 21,218 SNPs, with minor allele
frequencies in the sample of 8 fish greater than 0.25,
using Stacks software v~1.12 \citep{catchen2011stacks} with the 
following parameter settings ($m = 4$, $M = 2$, $n = 2$). At 1,118 of the SNPs the four Klamath
samples carried a single allele that was different from the single allele carried by the four
Sacramento samples.  96 of these SNPs with the highest read depths and quality scores  were
submitted to Fluidigm Inc. (South San Francisco, CA) for \snptype{} assay design.  Although
these SNPs appeared to be fixed for alternate alleles in the sequencing samples, many of them
were polymorphic in the reference samples (see Results).

The 96 \snptype{} assays are loaded together with 94 green sturgeon samples and
two non-template controls onto 96.96 Fluidigm Dynamic Arrays for genotyping. Fluidigm nanofluidic 
technology allows simultaneous genotyping of all 9,216 sample-by-locus combinations.  Each allele
at a SNP corresponds to a fluorescent dye and the genotype of an individual can be scored from the
intensity of fluorescence for each of those dyes.  Fluidigm provides software that allows computerized
scoring of diploid individuals, but the software does not allow scoring of tetraploid genotypes.
 

\subsection*{Developing a Scoring Methodology}
While there have been recent advances in methodologies for scoring the genotypes of
tetraploid individuals using SNP assay intensity data, when we
used one of these methods, {\sc FitTetra} \citep{voorrips2011genotype}, on our data, only 49 of the 96
SNPs yielded called
genotypes.  Further, upon visual inspection of the
intensity data for the called genotypes, we had high confidence in the reproducibility of the 
genotype calls in even fewer assays. And finally, {\sc FitTetra} does not appear to have an easy 
mechanism for using multiply-genotyped individuals to guide the SNP selection and intensity-modeling process.
Accordingly, we developed our own manual scoring procedure based on the intensity data from the 
Fluidigm platform.  This is manageable since we are using only 96 assays and have a relatively
small number of samples, and in our opinion, is a preferable strategy, for this project, to using the program
{\sc FitTetra}.  It should be noted that the recommended procedure for using {\sc FitTetra} involves a 
substantial decision-making effort by the user, including empirical 
adjustment of parameters and some hand curation. Our approach does the same, but with a more visual
method, rather than using the underlying modeling framework of {\sc FitTetra}.

Apparent in our data were two difficult features. First, at many of the 
green sturgeon SNPs there were not five, clear, scorable genotypes, as might be
hoped for in tetraploids; rather, many of the clusters in the fluorescence intensity data were 
indistinct or overlapping. Second, 
we encountered considerable chip to chip variation in fluorescence
intensity. To combat the first difficult feature of our data, we noted that
for population assignment
and for individual identification, it is not necessary to resolve the
exact dosage for each allele
within an individual.  It suffices to merely score an individual into a 
particular {\em genotype category} without reference to the underlying alleles.  Thus, individuals 
with different genotypes may be scored as being in the same genotype category if they are
not well resolved in fluorescent intensity.  This approach allows the user to define genotype
categories that are easily and clearly resolved. To address the chip-to-chip variation we 
include, on each chip, a number of individuals that have appeared on a previous chip, so that 
corresponding clusters between the chips can be identified. 

We first used four chips (our {\em scoring-development chips}, Table~\ref{tab:devo-samps}) to decide which loci were consistently scorable, and at each, the number of genotype  categories
that could be resolved.  Some individuals were genotyped on multiple chips which allowed us to visually account for 
chip-to-chip variation.  
\begin{table}
\hrule \kern 0.5mm \hrule
\small
\caption{Four scoring-development chips}
\label{tab:devo-samps}
(a) Number of samples from each location
\begin{center}
\begin{tabular}{lrrr}
\input{../outputs/chip_location_counts.tex}
\hline
\end{tabular}
\end{center}
\mbox{}\\
(b) Number of samples shared between chips
\begin{center}
\begin{tabular}{lrrrr}
\input{../outputs/cross_chip_shared_samples.tex}
\hline
\end{tabular}
\end{center}
\hrule
\end{table}
We created a series of plots (one for each locus) plotting the raw fluorescent intensities from the
Fluidigm chip reader.  The {\em raw} intensities are those that have not been transformed based on the
observed intensity of the two no-template controls (NTCs) on the chip.
Raw intensities are not available from the
Fluidigm SNP Genotyping Analysis Software by default, but, with a request and a code from
Fluidigm technical support (ph.~866-358-4354), they can be
accessed.  We used the raw intensities at this stage because there was less chip-to-chip variability in them
compared to the, NTC-normalized values available, by default, from the software. On these plots we colored the
results from each chip a different color and connected the same sample genotyped on different chips with
line segments.  The plots of four exemplar loci with 2, 3, 4, and 5 genotype categories regarded as scorable, respectively,
are shown in Figure~\ref{fig:plate_xy}. Similar plots for all of the loci are available in the Supporting
Information.
\begin{figure*}
(a)\\
\includegraphics[width = \textwidth]{../outputs/four_loci_by_plate-crop.pdf} \\
(b)\\
\includegraphics[width = \textwidth]{../outputs/four_loci_by_geno-crop.pdf}
\caption{\footnotesize Raw \snptype{} fluorescence intensities at four loci typed upon the four
scoring-development chips. Each panel
shows the results at four chips for one locus.  The numeral in the upper right gives the number
of genotype categories that are scored at the locus.  Each point represents an individual's fluorescence
intensities at two dyes. ({\em a}) Individuals are colored according to chip and dashed line
segments connect the intensities of individuals typed on different chips. ({\em b}) Individuals are colored
according to the genotype category to which they have been scored.  Category~0 (colored light gray) is a ``no call'', \ie the individual
is recorded as having missing data at the locus.}
\label{fig:plate_xy}
\end{figure*}
It is particularly instructive to look at locus ame\_21745--- the first panel in Figure~\ref{fig:plate_xy}(a).    
At this locus there are three clusters of points that might be resolved; however the two clusters on the right side of the plot 
are rather close together and could be difficult to resolve in the face of substantial chip-to-chip variation.   Accordingly,
we do not attempt to score three separate genotype categories at this locus, choosing instead to score only two categories. 
In general, we tended to be conservative, preferring to merge nearby clusters into a single genotype category, 
ensuring that genotype categories could be called reliably, rather than maintaining more genotype categories,
some of which could be subject to high rates of genotyping error.

In order to score more chips beyond the initial four scoring-development-chips we follow a simple routine. 
First, we include between 16 and 24 individuals from the scoring-development-chips on the new chip. 
Second, we plot the raw intensities of the new chip along with the raw intensities on the four scoring-development-chips, 
including line segments connecting the 16 to 24 individuals between the old and the new chip intensities. 
This provides visual verification of the correspondence between clusters on the new chip and the genotype
categories chosen on the basis of the four scoring-development-chips. With those plots as a helpful guide, 
we then score the new chip using the Fluidigm software. Because this software only allows individuals to be
manually scored into three genotype categories (one, two, and three), we overload those three categories. In other words, if we
score a genotype as a one (or a two) but its intensities are such that it falls in the upper left part of 
the X-Y plot, then, in post-processing the output of the Fluidigm software, we convert that score to 
genotype category four (or five).  If the point fell in the lower right part of the plot, 
then it would have remained in genotype category one (or two).  To ensure that scoring can be done easily
by different members of our lab, and to allow other labs to use this scoring procedure, we 
will wrap the necessary data and R code into an R package, {\tt gsSNPscore}.

\subsection*{Population structure,  population assignment, and individual assignment}
Our scoring method does not resolve the exact genotype of each individual---the precise dosage of different alleles within an individual is not obtained. Accordingly, these data are not appropriate for use in analytical models that assume information 
about the individual, constituent alleles of each individual's genotype. However, 
there are still a number of statistical analyses that may be done in an entirely 
valid fashion using only the genotype category data that we have collected. 
Notably, applying the program {\em structure} assuming a haploid organism and
using the model {\em without} admixture is 
entirely valid.  In that case, the software implements a finite mixture model in which 
the component-specific parameters are the frequencies of different genotype categories
at each locus, and an individual's genotype category is assumed drawn, independently 
at each locus, from these frequencies.  
Such a model does not require any notion of
constituent alleles or mode of inheritance.    We included all individuals from which we
had successfully obtained scores at 60 or more loci in multiple runs
of {\em structure} at $K$ (the number of subpopulations) equal to one, two, and three.

In similar fashion, population assignment or, as
it is often called in fisheries, ``genetic stock identification'' (GSI), can be carried out, using
our definition of genotype categories, under
a haploid model.  The model underlying GSI is nearly identical to the {\em structure} model described above,
except that {\em structure} assumes that the proportion of individuals from each component is
equal, while in GSI that proportion is estimated.  We used the egg/juvenile reference samples from the Sacramento and Klamath
rivers, as our "baseline" samples from the Southern and Northern DPSs, respectively.
We then performed GSI on the remaining samples, using the program {\em gsi\_sim}
\citep{Andersonetal2008,Israeletal2009},
obtaining for each fish a posterior probability that it belongs to the Southern or Northern DPS.


Though it could be challenging to use our genotype categories, which have unknown inheritance properties,
to infer related individauls (\ie to identify parent-offspring pairs, or full-siblings), it is possible
to use the observed genotype categories to identify the same individual, multiply-sampled.  We developed a
straightforward likelihood-based approach to identify genotypes from the same sample.  Let $\bp_\ell$ 
be the frequencies of the genotype categories at locus $\ell$ in the DPS from which a pair of individuals
originate.  For example, if four categories are 
scored at locus $\ell$, then $\bp = (p_{\ell,1}, \ldots, p_{\ell,4})$, with $\sum_{k=1}^4 p_{\ell,k} = 1$.
Assume that at locus $\ell$ a fraction $\epsilon_\ell$ of the samples is expected to be subject to 
genoytping error.  If a sample is subject to genotyping error, then its observed genotype category
is drawn from $\bp_\ell$, independently of its true, unobserved genotype category. (Though this is
a somewhat unrealistic genotyping error model, it is very convenient, mathematically and adequate for
our purposes.)  Under this assumption,
the marginal probability that a sample, $i$, is observed to have genotype category $g$ at locus $\ell$ is
\[
P(y^{(i)}_\ell = g) = (1-\epsilon)p_{\ell,g} + \epsilon p_{\ell,g} = p_{\ell,g}.
\]
If another sample $j$ that is unrelated to sample $i$---we shall say $K_{ij} = \mathrm{U}$---is typed
and found to have genotype category $h$ (which may be the same category as $g$) then the joint probability is simply
\begin{eqnarray}
P(y^{(i)}_\ell = g, y^{(i)}_\ell = h | K_{ij} = \mathrm{U}) &=& P(y^{(i)}_\ell = g)P(y^{(j)}_\ell = h)  \nonumber \\
& = & p_{\ell,g} p_{\ell,h}  \label{eq:pairU}
\end{eqnarray}
If $i$ and $j$ are two samples from the same individual ($K_{ij} = \mathrm{S}$), then the joint probability of the observed genotypes
can be written as a sum over the four possible cases: 1) neither sample suffered a genotyping error; 2) $i$ suffered
an error, but not $j$; 3) $j$ suffered an error, but not $i$; 4) both $i$ and $j$ suffered an error:
\begin{eqnarray*}
\lefteqn{P(y^{(i)}_\ell = g, y^{(i)}_\ell = h | K_{ij} = \mathrm{S})  =} \\
& & (1-\epsilon_\ell)^2  p_{\ell,g}  \delta(g = h)  \\
& & \mbox{} + \epsilon_\ell p_{\ell,g} \times (1-\epsilon_\ell) p_{\ell,h}  \\
& & \mbox{} + (1-\epsilon_\ell)p_{\ell,g} \times \epsilon_\ell p_{\ell,h} \\
& & \mbox{} + \epsilon_\ell^2 p_{\ell,g} p_{\ell,h},
\end{eqnarray*}
where $\delta(g = h)$ is 1 when $g=h$ and 0 otherwise.  This can be written more compactly as
\begin{eqnarray}
\lefteqn{P(y^{(i)}_\ell = g, y^{(i)}_\ell = h | K_{ij} = \mathrm{S})  =} \\ \nonumber
& & (1-\epsilon_\ell)^2  p_{\ell,g}  \delta(g = h) + (2\epsilon_\ell - \epsilon_\ell^2) p_{\ell,g} p_{\ell,h}. \label{eq:pairS}
\end{eqnarray}

A log-likelihood ratio statistic that can be used to identify pairs of samples $i$ and $j$ from the same
individual is then
\begin{equation}
\Lambda_{ij}= \log\frac{P(y^{(i)}_\ell = g, y^{(i)}_\ell = h | K_{ij} = \mathrm{S})}
{P(y^{(i)}_\ell = g, y^{(i)}_\ell = h | K_{ij} = \mathrm{U})}
\label{eq:logl}
\end{equation}
To approximate the distribution of $\Lambda_{ij}$ under $K_{ij} = \mathrm{S}$ and $K_{ij} = \mathrm{U}$, we
conducted Monte Carlo simulations using values of $\bp$ for the Northern and Southern DPSs obtained
by counting genotyping categories scored on individuals assigned by {\sc structure} to the Northern DPS and Southern DPS
clusters.  For the simulation of pairs and for the calculation of $P(y^{(i)}_\ell = g, y^{(i)}_\ell = h | K_{ij} = \mathrm{S}$ 
for each simulated pair we assumed a conservatively high genotyping error rate of $\epsilon_\ell = 0.05$ for every locus.  
$10^5$ genotype pairs were simulated both for $K_{ij} = \mathrm{S}$ and $K_{ij} = \mathrm{U}$. Simulations were first done for 
the case in which every pair was scored at all of the SNPs developed in our
panel, and we successively deleted loci randomly from each simulated pair to investigate the expected
distributions when pairs shared fewer scored loci due to missing data (\ie loci that were not scored).
We computed $\Lambda_{ij}$ for all pairs
$ij$ of fish that were either assigned to the Northern DPS or the Southern DPS. Both the
simulations and the pairwise comparisons of real fish were done using a custom-written
script in the R programming language \citep{Rcore2015}.



\section*{Results}

\subsection*{Loci scored}
Of the 96 SNP assays we originally developed, we found that 74 of them could be reliably scored 
into genotype clusters. Of these, \Sexpr{num_loc_by_num_cat["2"]} were loci with two genotype
categories, \Sexpr{num_loc_by_num_cat["3"]} with three, \Sexpr{num_loc_by_num_cat["4"]} with four,
and \Sexpr{num_loc_by_num_cat["5"]} with five.  Of the 470 attempts made to genotype a DNA sample 
on the four scoring development chips and chip~five, \Sexpr{num_60_locs_or_more} of those yielded scored genotype
categories in at least 60 of the 74 loci. Inspection of the distribution of the number of scored 
loci revealed that, most sample types were consistently genotyped well ($\geq 60$ scored genotype categories)
except for the egg samples from the Sacramento River (Fig~\ref{fig:success-histos}).


\begin{figure}
\begin{center}
\includegraphics[width = \linewidth]{../outputs/successful-assay-histogram-crop.pdf}
\end{center}
\caption{\footnotesize Histogram showing the number of individuals
successfully scored for $0,\ldots, 74$ assays, on each of the chips.  Color denotes type of sample; Group Short
Names as in Table~\ref{tab:samps}.  \label{fig:success-histos}}
\end{figure}



\subsection*{Analysis with {\em structure}}
324 individuals had scored genotype categories at 60 or more loci, and were hence included in the analysis with 
{\em structure.}  This analysis was a clustering analysis in which no prior information was given about the origin
of the samples, even if that was known (as is the case with the reference samples.)  The posterior probability that each
individual originated from one of the $K$ clusters is given in Figure~\ref{fig:distruct}.
%%%%%%
\begin{figure*}
\includegraphics[width = 0.92\textwidth]{../StructureArea/clump_and_distruct/final_pdf/BB_ds_Clumped_TopLabel_k002r001.pdf}~\raisebox{0.96ex}{$K = 2$} \\
\includegraphics[width = 0.92\textwidth]{../StructureArea/clump_and_distruct/final_pdf/BB_ds_Clumped_NoLabel_k003r001.pdf}~\raisebox{0.96ex}{$K = 3$} \\
\includegraphics[width = 0.92\textwidth]{../StructureArea/clump_and_distruct/final_pdf/BB_ds_Clumped_NoLabel_k004r001.pdf}~\raisebox{0.96ex}{$K = 4$}
\caption{\footnotesize Representative {\em distruct} \protect\citep{rosenberg2004distruct} plots of {\em structure} estimates 
of posterior probabilities of cluster membership for \Sexpr{N_for_structure} individuals with $\geq 60$ loci. Sampling location 
codes in Table~\ref{tab:samps}. Each individual is represented by a vertical bar and the amount of different colors in those
bars represents the posterior probability of cluster membership. For example, with $K=2$ each bar is either completely orange (Northern 
DPS cluster) or completely blue (Southern DPS cluster), indicating that individuals can be assigned to those clusters with effectively 
no uncertainty. For $K=3$ and $K=4$ there is more noise, indicating that a model with more than two subpopulations of green sturgeon 
is not supported by the genetic data.\label{fig:distruct}}
\end{figure*}
%%%%%
The results from {\em structure} show unambiguously that the model for population structure that is best supported by the data
is one with $K=2$ clusters, and noting the clusters to which the reference samples are assigned it is clear that the clusters
defined by structure coincide perfectly with the Northern and Southern DPSs of green sturgeon.  The bycatch consists of fish that
are from both the Northern and Southern DPSs, with a greater proportion from the Southern DPS.  Likewise, it is clear that the 
fish sampled from the Eel River belong to the Northern DPS.

\subsection*{Population assignment with {\em gsi\_sim}}


\subsection*{Individual identification}
\begin{figure*}
{\bf (a)}\\
\includegraphics[width = \textwidth]{../outputs/lambda_densities-crop.pdf}\\
{\bf (b)}\\
\includegraphics[width = \textwidth]{../outputs/obs-self-id-logls-crop.pdf}
\caption{\footnotesize Simulated and observed distribution of self-ID log likelihood ratio statistics.
Remember to note that the simulations were done assuming a high genotyping error rate.}
\end{figure*}

\textcolor{red}{Note, there seems to be a clear error in the duplicate samples page in the spreadsheet.  Namely, AM000120 and AM000149 are clearly the same fish as are AM000131 and AM000137, but the sample sheet has that AM000120 and AM000131 are the
same and that AM000137 and AM000149 are the same.}
\section*{Discussion}

\section*{Acknowledgments}
We are grateful to \ldots
%%%%%%%%%%%%%%%% END MAIN BODY
\bibliographystyle{men}
{\footnotesize
\bibliography{sturgeon-fluidigm}}

 \end{document}


