\documentclass[twoside,10pt,twocolumn]{article}

% this stuff just has to be in here so it doesn't choke 
% on the stuff that I put in for typesetting the submission version.
\newcommand{\AuthorAddresses}{}
\newcommand{\KeyWords}{}
\newcommand{\CorrespondingAuthor}{}
\newcommand{\RunningTitle}{}


\input{author-title-etc}  % Put all the author title stuff in there

% ------
% Fonts and typesetting settings
\usepackage[sc]{mathpazo}
\usepackage[T1]{fontenc}
\linespread{1.05} % Palatino needs more space between lines

\usepackage{microtype}
% ------
% Page layout
\usepackage[labelfont=bf,labelformat=simple,labelsep=quad]{caption}

\usepackage{graphicx}
\usepackage{amssymb}
\usepackage{epstopdf}
\usepackage{amsfonts}
\usepackage{natbib}
\usepackage{subfigure}
\usepackage{xspace}
\usepackage{mathrsfs}
\usepackage{fancyhdr}
\usepackage{cuted}
\usepackage{flushend}
\usepackage{color}



%% some handy things for making bold math
\def\bm#1{\mathpalette\bmstyle{#1}}
\def\bmstyle#1#2{\mbox{\boldmath$#1#2$}}
\newcommand{\thh}{^\mathrm{th}}


%% Some pretty etc.'s, etc...
\newcommand{\cf}{{\em cf.}\xspace }
\newcommand{\eg}{{\em e.g.},\xspace }
\newcommand{\ie}{{\em i.e.},\xspace }
\newcommand{\etal}{{\em et al.}\ }
\newcommand{\etc}{{\em etc.}\@\xspace}

\newcommand{\snptype}{SNP~Type\textsuperscript{{\tiny TM}}}


%% the page dimensions
\textwidth = 6.5 in
\textheight = 9 in
\oddsidemargin = -0.01 in
\evensidemargin = -0.01 in
\topmargin = -0.7 in
\headheight = 0.25 in
\headsep = 0.25 in
\parskip = 0.0in
\parindent = 0.25in

\setlength{\columnsep}{.4in}


%% change section heading styles
\makeatletter
\renewcommand\section{\@startsection {section}{1}{\z@}%
                                   {-3.5ex \@plus -1ex \@minus -.2ex}%
                                   {1.5ex \@plus 0ex}%
                                   {\normalfont\normalsize\bfseries}}
\renewcommand\subsection{\@startsection{subsection}{2}{\z@}%
                                     {-3.25ex\@plus -1ex \@minus -.2ex}%
                                     {1.5ex \@plus .2ex}%
                                     {\normalfont\normalsize\itshape}}
\makeatother

%% modify the abstract environment
\renewenvironment{abstract}
{\begin{quote}{\bf Abstract}\end{quote}\begin{quote}\bfseries\small}
{\end{quote}}


\fancypagestyle{firststyle}
{
   \fancyhf{}
   \chead[]{\vspace*{.1in}\includegraphics[width=\textwidth]{images/banner.pdf}}
   \lfoot[]{\footnotesize \myCopyright}
}

%% here is what I hope will be fancyplain
\fancyhead{} % clear all header fields
\fancyhead[LE]{{\bf \thepage}~~{\sl \myRunningTitle}}
\fancyfoot[RE,LO]{{\footnotesize \myCopyright}}
\renewcommand{\headrulewidth}{0pt}
\fancyhead[RO]{{\sl \myRunningAuthor}~~{\bf \thepage}}
\cfoot[]{}


%%%%%%%% KNITR BLOCK
% Here is a knitr chunk to set global values
<<setup, include=FALSE, cache=FALSE>>=
docDir <- getwd()

# set global chunk options
knitr::opts_chunk$set(include = FALSE, cache = FALSE)
knitr::opts_knit$set(root.dir = file.path(docDir, ".."))
#knitr::opts_knit$set(self.contained = FALSE)
knitr::opts_knit$set(concordance = TRUE)
@



% a knitr chunk to evaluate the code needed to produce variables and figures we need
% in this section
<<meta-data-summarise, include=FALSE, cache = FALSE>>=
source("R-main/00-meta-data-summaries.R")
@



\begin{document}


\pagestyle{fancyplain}
\thispagestyle{firststyle}



   \begin{strip}
   \mbox{}\\
        {\LARGE\bf \myTitle \par}
    \mbox{}\\
    \uppercase{\myAuthors}\\ 
       \mbox{}\\
    {\em \myAffiliations}\\
    \mbox{}\\
    {\small \myEmailFootnote Correspondence: \myEmailAddress}
    
   \begin{abstract}
   	\input{abstract.Rnw}     %% Put abstract text in here.  If we want to include R code in it we will have to
                             %% use Sexpr{knit_child("doc.Rnw")} instead of \input{abstract.Rnw}.  But using \input gives us good
                             %% syncing
    \end{abstract}
   \end{strip}

{\small
% here, input the main body text part.
%<<main-body-insert, child="main-body-text.Rnw">>=
%@
%% I grew weary of not having PDF sync, so I just inserted the whole file in here.  That
%% Makes it harder to do the one column version, but I can probably figure out a workaround
%% By just processing the file with awk and putting this into it.
%% It is a shame that Yihui doesn't want to support concordance with child documents.
%%%%%%%%%%%%%%%% START MAIN BODY

\section*{Introduction}

This is super free-form, stream of consciousnessy at the moment, as I get some ideas
down onto the page.

The study of highly migratory species in environments that are difficult to monitor present
a number of challenges.   Often, the migrations themselves cannot be observed.
Genetic data has been successful in inferring the ecology of a number of species
that are otherwise difficult to study (CITE).  It is now fairly routine to do this
in most species, especially diploid ones.  Genetic methodologies are well-established for microsats, SNPs, and to an increasing extent, lately, next generation sequencing data.
Concurrently, the last two decades have seen an explosion in the availability statistical
methods for analyzing genetic data.  

The vast majority of methods have been developed for diploid organisms.  When these methods
can be extended to polyploids their extension
typically requires that the number of copies of individual alleles can be resolved in a
polyploid genotype, or that the exact inheritance patterns are known, which is not always
the case.  The challenges of polyploids have been
well-appreciated in botanical fields for some time (CITE Fisher's polyploid inheritance
paper), and these difficulties have been gaining increasing attention in the field of
molecular ecology (CITE that recent review).  

Significant advances have been made in the development of models for polyploids and
particularly for tetraploids, for example, fitTetra, and the work from Brazil on
sugar cane.  However, sometimes that quality of scoring is not high enough to 
resolve all those allele copies. Furthermore, the assumptions underylying fitTetra
are probably not appropriate when samples are mixture of diverged tetraploid populations.
(ERIC VERIFY THIS)
Clearly, it would be best to be able to resolve
and count all the alleles, but in cases where that is not possible for a majority of the
loci that have been discovered, alternative modes of analysis are sought.  One such
alternative analysis method is to treat alleles in polyploids as dominant markers.  This
approach has been used recently in population assignment (Israel et al.) and CITE, and CITE,
and more recently Wang and Scribner have shown the utility of that approach for doing
relationship inference.  

Modeling everything as a dominant marker (presence/absence of the allele) is particularly
well-suited to microsatellite loci where length polymorphisms are easy to interpret.
However, using xxxxxx-type SNP assays (like fluidigm assays)  the individual alleles
may not be so easily resolved, because the scoring relies on individual genotypes clustering
in two space, and it might not always be easy to resolve the 5 expected genotype clusters.
For such data, trying to resolve the data into individual, expected genotype categories
might be unworkable for many loci, especially if typing members of fairly well-diverged
lineages.  Depending on the goal of genetic analysis, however, it may still not be
necessary to require thinking about the data only as  individual alleles giving rise to
a full complement of expected genotypes.



\section*{Methods}



\subsection*{Genetic Samples}

We received samples of two general categories.  Samples collected in freshwater rivers known to have 
well-established spawning populations of green sturgeon---Sacramento River and Klamath River, in this study---are
``reference'' samples. Some of these samples were taken from eggs collected on egg mats or 
from juvenile fish, either of which must have been spawned in the river from which they were sampled. 
We refer to these as ``egg/juvenile reference'' samples.  By contrast, ``non-juvenile reference'' samples are 
taken from adult fish that may be non-spawning visitors that are actually native to a river different 
from the one they were sampled in. 
Samples taken from the ocean or from rivers without an historically abundant 
spawning population are ``non-reference'' samples.  They are comprised of green sturgeon caught as 
bycatch, and five adult fish sampled from the Eel River, CA.   See Table~\ref{tab:samps} for a summary
of sample sizes.
\begin{table*}
\caption{Summary of genetic samples used in this study.}
\begin{center}
\begin{tabular}{llllr}
\hline \hline \\ 
\input{../outputs/sample_summary.tex}
\hline \hline \\ 
\end{tabular}
\end{center}
\label{tab:samps}
\end{table*}
Tissue samples were predominantly fin clips dried or stored in ethanol, apart from the eggs which were whole developing eggs, stored in ethanol, and in the case of the Klamath River reference samples, whole, young juveniles in ethanol. 



\subsection*{SNP Discovery and Assay Development}

To discover SNPs to be made into Fluidigm \snptype{} assays,  we first sequenced eight fish
using a double-digest restriction-site associated DNA protocol
(ddRAD; \citealt{Petersonetal2012})  on a MiSeq sequencer (Illumina Inc., San Diego, CA).
This protocol uses restriction enzyme digestion and DNA fragment size selection
to reduce the fraction of
genome that is sequenced, ensuring that a small portion of the genome can be sequenced to
a sufficiently high read depth in a number of different individuals.  Four of the samples were
juveniles from the Klamath River (Northern DPS) and four were juveniles (????) from the
Upper Sacramento River. 21.24 million sequence reads of
XXXX base pairs generated by the sequencing run were assembled into 84,747 putatively unique genetic 
loci (regions of the genome) and, within those regions, we identified XXX,XXX SNPs
using Stacks software v 1.12 \citep{catchen2011stacks} with the 
following parameter settings ($m = XX$, $M = XX$, XXXX). At 1,118 of the SNPs the four Klamath
samples carried a single allele that was different from the single allele carried by the four
Sacramento samples.  96 of these SNPs with the highest read depths and quality scores  were
submitted to Fluidigm Inc. (South San Francisco, CA) for \snptype{} assay design.  Although
these SNPs appeared to be fixed for alternate alleles in the sequencing samples, many of them
were polymorphic in the reference samples (see Results).

The 96 \snptype{} assays are loaded together with 94 green sturgeon samples and
two non-template controls onto 96.96 Fluidigm Dynamic Arrays for genotyping. Fluidigm nanofluidic 
technology allows simultaneous genotyping of all 9,216 sample-by-locus combinations.  Each allele
at a SNP corresponds to a fluorescent dye and the genotype of an individual can be scored from the
intensity of fluorescence for each of those dyes.  Fluidigm provides software that allows computerized
scoring of diploid individuals, but the software does not allow scoring of tetraploid genotypes.
 

\subsection*{Developing a Scoring Methodology}
While there have been recent advances in methodologies for scoring the genotypes of
tetraploid individuals using SNP assay intensity data, when we
used one of these methods, {\sc FitTetra} \citep{voorrips2011genotype}, on our data, only 49 of the 
SNPs yielded called
genotypes, which is insufficient for individual identification.  Further, upon visual inspection of the
intensity data for the called genotypes, we had high confidence in the reproducibility of the 
genotype calls in even fewer assays.
Accordingly, we developed our own manual scoring procedure based on the intensity data from the 
Fluidigm platform.  This is manageable since we are using only 96 assays and have a relatively
small number of samples, and in our opinion, is a preferable strategy to using the program
{\sc FitTetra}.  It should be noted that the recommended procedure for using {\sc FitTetra} involves a 
substantial decision-making effort by the user, including empirical 
adjustment of parameters and some hand curation. Our approach does the same, but with a more visual
approach, rather than using the underlying modeling framework of {\sc FitTetra}.

Apparent in our data were two difficult features. First, at many of the 
green sturgeon SNPs there were not five, clear, scorable genotypes, as might be
hoped for in tetraploids; rather, many of the clusters in the fluorescence intensity data were 
indistinct or overlapping. Second, 
we encountered considerable chip to chip variation in fluorescence
intensity. To combat the first difficult feature of our data, we noted that
for population assignment
and for individual identification, it is not necessary to resolve the
exact dosage for each allele
within an individual.  It suffices to merely score an individual into a 
particular {\em genotype category} without reference to the underlying alleles.  Thus, individuals 
with different genotypes may be scored as being in the same genotype category if they are
not well resolved in fluorescent intensity.  This approach allows the user to define genotypes
categories that are easily and clearly resolved. To address the chip-to-chip variation we 
include, on each chip, a number of individuals that have appeared on a previous chip, so that 
corresponding clusters between the chips can be identified. 

We first used four chips (our {\em scoring-development chips}) to decide which loci were consistently scorable, and at each, the number of genotype clusters
that could be resolved.  Some individuals were genotyped on multiple chips which allowed us to visually account for 
chip-to-chip variation (Table~\ref{tab:devo-samps}).  
\begin{table}
\hrule \kern 0.5mm \hrule
\small
\caption{Four scoring-development chips}
\label{tab:devo-samps}
(a) Number of samples from each location
\begin{center}
\begin{tabular}{lrrr}
\input{../outputs/chip_location_counts.tex}
\hline
\end{tabular}
\end{center}
\mbox{}\\
(b) Number of samples shared between plates
\begin{center}
\begin{tabular}{lrrrr}
\input{../outputs/cross_chip_shared_samples.tex}
\hline
\end{tabular}
\end{center}
\hrule
\end{table}
We created a series of plots (one for each locus) plotting the raw fluorescent intensities from the
Fluidigm chip reader.  The {\em raw} intensities are those tht have not been transformed based on the
observed intensity of the two no-template controls (NTCs) on the chip.
Raw intensities are not available from the
Fluidigm XXXX Software by default, but, with a request and a code from
Fluidigm technical support (ph.~XXX-XXX-XXXX), they can be
accessed.  We used the raw intensities at this stage because there was less chip-to-chip variability in them
compared to the standard, NTC-normalized values available from the software. On these plots we colored the
results from each chip a different color and connected the same sample genotyped on different chips with
line segments.  An example for four loci with 2, 3, 4, and 5 clusters regarded as scorable, respectively,
is shown in Figure~\ref{fig:plate_xy}. Similar plots for all of the loci are available in the Supporting
Information.
\begin{figure*}
(a)\\
\includegraphics[width = \textwidth]{../outputs/four_loci_by_plate-crop.pdf} \\
(b)\\
\includegraphics[width = \textwidth]{../outputs/four_loci_by_geno-crop.pdf}
\caption{\footnotesize Raw \snptype{} fluorescence intensities at four loci typed upon the four
scoring-development chips. Each panel
shows the results at four plates for one locus.  The numerial in the upper right gives the number
of genotype categories that are scored at the locus.  Each point represents an individual's fluorescence
intensities at two dyes. (a) Inidividuals are colored according to plate and dashed line
segments connect the intensities of individuals typed on different chips. (b) Individuals are colored
according to the genotype category they have been scored as.  Category~0 (colored light gray) is a ``no call'', \ie the individual
is recorded as having missing data at the locus.}
\label{fig:plate_xy}
\end{figure*}


at a SNP are are called by Resultant raw fluorescence data from samples of known origin were submitted to FitTetra package for R \citep{voorrips2011genotype}, which fits mixture models to call tetraploid loci, allowing us to call genotypes in 49 of the 96 loci. Of these 49 loci, 29 had genetic differentiation greater than 1\% between known samples from the Northern and Southern DPS as measured by the GST statistic in Genodive v2.0b25 \citep{meirmans2004genotype}. We then manually curated the genotype data, selecting 12 loci for use in Genetic Stock Identification based on their relative ease of scoring. Genotypes for these 12 loci from 21 Northern DPS and 39 Southern DPS individuals comprise the genetic baseline of known samples that we used in subsequent GSI analyses of bycatch. Raw fluorescence data from 152 bycatch samples were then scored using the mixture discriminant analysis function in the mda package for R \citep{hastie1996discriminant}, followed by manual curation. Genotype data from each of two sets of bycatch samples were submitted as separate fisheries mixtures to gsisim \citep{Andersonetal2008} for population assignment against the genetic baseline.

\section*{Results}

\section*{Discussion}

\section*{Acknowledgments}
We are grateful to \ldots
%%%%%%%%%%%%%%%% END MAIN BODY
\bibliographystyle{men}
{\footnotesize
\bibliography{sturgeon-fluidigm}}

 \end{document}


